HW_2/task_3.py

```diff
@@ -0,0 +1,46 @@
+# 3. Напишите программу, которая принимает две строки вида “a/b” - дробь с числителем и знаменателем.
+# Программа должна возвращать сумму и произведение* дробей.
+# Для проверки своего кода используйте модуль fractions.
+import fractions
+
+
+def multi_fraction(n1, n2):
+    b1 = list(map(int, n1.split('/')))
+    b2 = list(map(int, n2.split('/')))
+    a = b1[0] * b2[0]
+    b = b1[1] * b2[1]
+    for i in range(a, 1, -1):
+        if a % i == 0 and b % i == 0:
+            a = int(a / i)
+            b = int(b / i)
+
+    res = f'{a}/{b}'
+    return res
+
+
+def sum_fraction(k1, k2):
+    b1 = list(map(int, k1.split('/')))
+    b2 = list(map(int, k2.split('/')))
+    c1 = b1[1]
+    c2 = b2[1]
+    while c2 % c1 != 0:
+        c2 += b2[1]
+    c = int(b1[0] * (c2 / b1[1]) + (b2[0] * (c2 / b2[1])))
+    for i in range(c, 1, -1):
+        if c % i == 0 and c2 % i == 0:
+            c = int(c / i)
+            c2 = int(c2 / i)
+
+    res = f'{c}/{c2}'
+    return res
+
+
+a1 = '187/57'
+a2 = '169/1479'
```

Comment:
Использовал такие входные данные.
```suggestion
a1 = '1/2'
a2 = '1/2'
```
Вот что получил. С математической точки зрения, 1 / 1  будет равно 1, но хотелось бы получить точную копию отображения модуля `factions`.
![image](https://github.com/Maxis8/HW_Python/assets/49741194/00e3eacb-bf3a-43f9-8073-92b662a6923c)

Reply:
Вот еще
```suggestion
a1 = '0/2'
a2 = '1/2'
```
![image](https://github.com/Maxis8/HW_Python/assets/49741194/dbb6f07d-35bb-4706-add8-7642604c72dd)